LBHackney-IT/lbhackney-it.github.io
# Use Docusaurus to build

This PR replaces the static HTML used for the homepage of https://playbook.hackney.gov.uk with a Docusaurus static site that looks _similar_, so we can reconcile all the disparate documentation sites that make up the site into a single repository and build.

There will be separate PRs to migrate the content across from each repository, but I've kept them out of this PR to make review easier. As a result the base page/site you're reviewing doesn't have much/any actual content.

## What's changed?

1. The existing HTML content has been removed.
2. A new Docusaurus site has been added with the Hackney colour scheme and logos from the old content.
3. Deployment changes from classic GitHub pages to a new GitHub Action that publishes to GitHub Pages.
4. In preparation for the content migration, there are new `CONTRIBUTING` and `README` docs which document the architecture and how to make changes.

## Why is it necessary?

We have historically had many different static sites that are built independently to provide portions of the Hackney Development System site. This has several downsides:

1. There are many repositories to keep up to date. The content is split across at least 6 repositories. Most of these are Docusaurus sites with many JavaScript dependencies. These need to be updated independently and there is drift between versions and configuration across the different repositories. **This change reduces the maintenance burden to 1 repository/system, and eliminates the change of drift between sections**.
2. Each micro-site is isolated. Counterintuitively, the search and home links in the nav bar for each portion of the site go to the root of that portion. This makes navigation frustrating and it's difficult to search across all our documentation from one place. **This change consolidates everything in one place with consistent navigation and search experience across sections**.
3. Adding new content is _hard_. To add a new section a new micro-site repository must be spun up and configured. This is non-trivial because the sites rely on Algolia for search so a new index needs to be used. The Algolia account details may have been lost. **This change makes it trivial to add a new section (by adding a new directory). Search becomes local-only, removing the dependency on a 3rd party service.**
4. Restructuring content is even harder. The repository structure dictates the content structure, and moving content between these top level sections or rationalising them is prohibitively onerous because they're separate repositories. **This change makes it easy to move content and track restructures within a single git repository. Undoing those restructures also becomes trivial.**

This change was prompted by the need to add a joiners' process for GitHub and other GitHub management docs. The content didn't fit well within the existing structure and it felt disproportionate to create yet another repository for these docs.

## How to review this PR

1. Ignore  `package-lock.json` and the line change count! The PR looks intimidating with ~25k changes, but ~24.5k of them are the dependency list.
2. Run the site locally to see the changes and compare the homepage with https://playbook.hackney.gov.uk/. The links _are_ different, because we'll be linking to the consolidated content using Docusaurus magic links. As a result it'll look a bit sparse
3. Read the `README` and `CONTRIBUTING` docs. Do they make sense?
4. Check things like the favicon, title and search. We want to be sure the basics are working.
5. Walk through the commits. Each commit groups changes, so it's easy to review how we've added local search, Google Analytics, etc. this way. 

## How we'll merge this PR

If we merge this as-is then the homepage will be less useful for a while until we consolidate the content. We'll do a couple of things to mitigate that:

1. Review (& hopefully approve) this PR in isolation, so we catch anything wrong with the static site generator. 
2. Wait until the content migration PR is approved
3. Update the [GitHub Pages deployment source](https://github.com/LBHackney-IT/lbhackney-it.github.io/settings/pages) to be `GitHub Actions`.
4. Rename `master` to `main` just before merging (this will disable GitHub Pages until another commit is pushed)
5. Merge this PR
6. Immediately after merging this PR, merge the content migration PR, so there is only a tiny blip without the content available.

## Before this PR

![Screenshot 2024-10-16 at 15 06 (Brave Browser) 3@2x](https://github.com/user-attachments/assets/862c6371-ebce-4d5b-a001-a8c700e2325b)

## After this PR

![Screenshot 2024-10-16 at 15 06 (Brave Browser) 2@2x](https://github.com/user-attachments/assets/fcabb741-40ce-407c-845b-dc1b00d81425)
